### DISCUSSION TITLE
[KerasNLP]: Bangla News Article Classification

### DISCUSSION THREAD

**Discussion by innat** [keras-team/keras-hub#264] [Show and tell]
![test](https://user-images.githubusercontent.com/17668390/178124022-8f68f0c6-72a8-4e8a-a50f-f43f047b4e0e.gif)

We have published a code example that is a demonstration of using **KerasNLP** for text classification in non-English and local languages. Here we choose Bangla as the local language (over **376,226 news articles**) and build a simple transformer model to solve a multiclass classification task where we classify Bangla news articles into 5 categories. We use **KerasNLP** to build a transformer model and use **Alibi** for *Integrated Gradient* to explain the model.

• Kaggle notebook: [Bangla News Article Classification](https://www.kaggle.com/code/ipythonx/kerasnlp-bangla-news-article-classification)

We compare our model's performance with the published results obtained in the original paper, which had a **0.96** precision. The original authors described many preprocessing steps performed on the dataset. However, we get **~0.99** precision after training the transformer model only for 10 epochs in **KerasNLP** without any heavy preprocessing.

**Comment by chenmoneygithub** [comment]
Awesome!! It's nice to see KerasNLP is helping your work!!

**Reply by innat** [reply]
@chenmoneygithub Thanks. 
Though for explaining the model, this time I need to use [alibi](https://github.com/SeldonIO/alibi) but it would be great to have relevant functionality from keras_nlp. I've raised a ticket regarding this, [HERE](https://github.com/keras-team/keras-nlp/issues/93).
